pages/_app.js

```diff
@@ -23,19 +22,50 @@ export default function App({ Component, pageProps }) {
   // States
   const [query, setQuery] = useState("");
   const [dictionaryQuery, setDictionaryQuery] = useState("");
-  const [wordList, setWordList] = useState(dummyData);
   const [searchResults, setSearchResults] = useState([]);
   const [dictionaryResults, setDictionaryResults] = useState([]);
 
   // Fetching from Dictionary
   const DictionaryURL = `/api/dictionary-search/${dictionaryQuery}`;
-  const { data, isLoading, mutate } = useSWR(DictionaryURL, fetcher);
+  const { data: dictionaryData, isLoading: dictionaryIsLoading } = useSWR(
+    DictionaryURL,
+    fetcher
+  );
+
+  // Fetching from database
+  const DatabaseURL = `/api/word-list/`;
+  const {
+    data: databaseData,
+    isLoading: databaseIsLoading,
+    mutate: databaseMutate,
+  } = useSWR(DatabaseURL, fetcher);
+
+  // Word List useState
+  const [wordList, setWordList] = useState([]);
```

Comment:
This is an unnecessary state. You are just using it to put the databaseData into the state. You should instead use databaseData directly and handle the undefined of that state correctly.